impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -155,19 +156,52 @@ class StrokeGenerator {
         cap_proc(vtx_builder, polyline.GetPoint(contour_end_point_i - 1),
                  cap_offset, scale, /*reverse=*/false);
       } else {
-        join_proc(vtx_builder, polyline.GetPoint(contour_start_point_i), offset,
-                  contour_first_offset, scaled_miter_limit, scale);
+        join_proc(vtx_builder, polyline.GetPoint(contour_start_point_i),
+                  offset.GetVector(), contour_first_offset, scaled_miter_limit,
+                  scale);
       }
     }
   }
 
+  /// @brief  Represents a ray in 2D space.
+  ///
+  ///         This is a simple convenience struct for handling polyline offset
+  ///         values when generating stroke geometry. For performance reasons,
+  ///         it's sometimes adventageous to track the direction and magnitude
+  ///         for offsets separately.
+  struct Ray {
```

Comment:
This could perhaps go in //impeller/geometry.

Reply:
Not opposed to throwing this in geometry. I'll do so in a follow-up. Also occurs to me that `Ray` is an inaccurate name for this... Should be something like `SeparatedVector2`.